broker/tasks/iam.py

```diff
@@ -79,3 +80,34 @@ def delete_server_certificate(operation_id: str, **kwargs):
         )
     except iam_commercial.exceptions.NoSuchEntityException:
         return
+
+
+@huey.retriable_task
+def delete_previous_server_certificate(operation_id: str, **kwargs):
+    operation = Operation.query.get(operation_id)
+    service_instance = operation.service_instance
+
+    operation.step_description = "Removing SSL certificate from AWS"
+    db.session.add(operation)
+    db.session.commit()
+
+    if service_instance.instance_type == "cdn_service_instance":
+        iam = iam_commercial
+    else:
+        iam = iam_govcloud
+
+    for certificate in Certificate.query.filter(
+        and_(
+            Certificate.service_instance_id == service_instance.id,
+            Certificate.id != service_instance.current_certificate_id,
+        )
+    ).all():
+        try:
+            iam.delete_server_certificate(
+                ServerCertificateName=certificate.iam_server_certificate_name
+            )
+        except iam_commercial.exceptions.NoSuchEntityException:
+            pass
+        db.session.delete(certificate)
```

Comment:
I was going to ask if this was doable within the loop as you'd be modifying a list/set of elements in place, but that's not actually the case - you're queuing things up for a transaction and committing below, so it should be fine! :-)

Reply:
right - the `db.session.delete()` says "queue up a delete operation for this object" then `db.session.commit()` does the actual delete.

Reply:
*(tents fingers)* Excellent.